src/storage/page_buffer.c

```diff
@@ -7272,56 +7275,63 @@ pgbuf_delete_from_hash_chain (THREAD_ENTRY * thread_p, PGBUF_BCB * bufptr)
 /*
  * pgbuf_lock_page () - Puts a buffer lock on the buffer lock chain
  *   return: If success, PGBUF_LOCK_HOLDER, otherwise PGBUF_LOCK_WAITER
- *   hash_anchor(in):
  *   vpid(in):
+ *   fetch_mode (in):
+ *   hash_anchor(in):
+ *   perf (in/out):
  *
  * Note: This function is invoked only when the page is not in the buffer hash
  *       chain. The caller is holding hash_anchor->hash_mutex.
  *       Before return, the thread releases hash_anchor->hash_mutex.
  */
-static int
-pgbuf_lock_page (THREAD_ENTRY * thread_p, PGBUF_BUFFER_HASH * hash_anchor, const VPID * vpid)
+static PGBUF_LOCK_MODE
+pgbuf_lock_page (THREAD_ENTRY * const thread_p, const VPID * const vpid, PAGE_FETCH_MODE fetch_mode,
+		 PGBUF_BUFFER_HASH * const hash_anchor, PGBUF_FIX_PERF * perf)
 {
+  PGBUF_LOCK_MODE pgbuf_lock_state = PGBUF_LOCK_UNKNOWN;
+
 #if defined(SERVER_MODE)
   PGBUF_BUFFER_LOCK *cur_buffer_lock;
-  THREAD_ENTRY *cur_thrd_entry;
   TSC_TICKS start_tick, end_tick;
   UINT64 lock_wait_time = 0;
 
-  /* the caller is holding hash_anchor->hash_mutex */
-  /* check whether the page is in the Buffer Lock Chain */
-
-  if (thread_p == NULL)
+  // TODO: more accurate comments
```

Comment:
these existing comments are non-intelligible for me

Reply:
```suggestion
```